### PR TITLE
Implement cursor.remove() lifecycle method.

### LINKED ISSUE #2391: Cursor component doesn't have implementation for remove() function
**Description:**

- A-Frame Version:
- Platform / Device:
- Reproducible Code Snippet or URL:

### PR DESCRIPTION
**Pull request by donmccurdy**
Fixes #2391.

### CODE REVIEW

File: src/components/cursor.js [hunk 1]
```diff
@@ -51,12 +51,35 @@ module.exports.Component = registerComponent('cursor', {
...
-    cursorEl.addEventListener('raycaster-intersection-cleared',
-                              bind(this.onIntersectionCleared, this));
+    canvas.addEventListener('mousedown', this.onMouseDown);
+    canvas.addEventListener('mouseup', this.onMouseUp);
+    cursorEl.addEventListener('raycaster-intersection', this.onIntersection);
+    cursorEl.addEventListener('raycaster-intersection-cleared', this.onIntersectionCleared);
+  },
+
+  remove: function () {
+    var cursorEl = this.el;
```
Comment (dmarcos) [inline src/components/cursor.js]: `var el = this.el` for consistency?

File: src/components/cursor.js [hunk 2]
```diff
@@ -51,12 +51,35 @@ module.exports.Component = registerComponent('cursor', {
       return;
     }
 
+    this.onMouseDown = bind(this.onMouseDown, this);
```
Comment (dmarcos) [inline src/components/cursor.js]: Comment at the top `Bind methods`?

File: src/components/cursor.js (hunk 1 above)
Comment (donmccurdy) [inline src/components/cursor.js]: Done.

File: src/components/cursor.js (hunk 2 above)
Comment (donmccurdy) [inline src/components/cursor.js]: Done.

**Comment by dmarcos** [comment]
Thanks!

**Pull request merged**